Inlining expression does not evaluate same, as would with an intermediate variable

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 10 Pro v19044 64-bit
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-5775C CPU @ 3.30GHz  

CC version: Error: exec failed (CreateProcess) with code 2: A rendszer nem találja a megadott fájlt.

 cmd: cc --version

getwd: D:\Documents\Git\Base64-Demos\v
vmodules: C:\Users\-\.vmodules
vroot: D:\Desktop\v
vexe: D:\Desktop\v\v.exe
vexe mtime: 2022-05-17 04:43:14
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 02c8a60

Git version: git version 2.35.1.windows.2
Git vroot status: Error: fatal: not a git repository (or any of the parent directories): .git
.git/config present: false
thirdparty/tcc: N/A

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg .\main.v`
```v

fn to_bit_string(i rune) []u8 {
	return if i == 0 {[]u8{}} else {"${i:b}".split("").map(it.u8())}
}

fn from_bit_string(bits []u8) rune {
	//a := bits.map(it.str()).join("").parse_uint(2, 8) or {0}
	return if bits == [] { 0 } else { bits.map(it.str()).join("").parse_uint(2, 8) or {0} }
}

fn main() {
	from_bit_string(to_bit_string(u32(100)))
}
```

**What did you expect to see?**

When the return value is set to the variable "a" the code compiles without any problems, but when it is inlined like in "from_bit_string" function, it throws this error


**What did you see instead?**
```
==================
C:/Users/Imre5/AppData/Local/Temp/v_0/main.10744468123574316078.tmp.c:7030: warning: cast between pointer and integer of different size
C:/Users/Imre5/AppData/Local/Temp/v_0/main.10744468123574316078.tmp.c:11756: error: ')' expected (got "_t3")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/gen/c/if.v
+++ b/vlib/v/gen/c/if.v
@@ -29,7 +29,8 @@ fn (mut g Gen) need_tmp_var_in_if(node ast.IfExpr) bool {
 							if left_sym.kind in [.array, .array_fixed, .map] {
 								return true
 							}
-						} else if stmt.expr.or_block.kind != .absent {
+						}
+						if stmt.expr.or_block.kind != .absent {
 							return true
 						}
 					}

--- a/vlib/v/tests/if_expr_with_method_call_optional_test.v
+++ b/vlib/v/tests/if_expr_with_method_call_optional_test.v
@@ -1,0 +1,13 @@
+fn to_bit_string(i rune) []u8 {
+	return if i == 0 { []u8{} } else { '${i:b}'.split('').map(it.u8()) }
+}
+
+fn from_bit_string(bits []u8) rune {
+	return if bits == [] { 0 } else { bits.map(it.str()).join('').parse_uint(2, 8) or { 0 } }
+}
+
+fn test_if_expr_with_method_call_optional() {
+	a := from_bit_string(to_bit_string(u32(100)))
+	println(a)
+	assert a == `d`
+}

